lua/wikis/valorant/MatchGroup/Input/Custom.lua

```diff
@@ -137,6 +157,161 @@ function MatchFunctions.getPatch(match, games)
+				and map.extradata
+				and map.extradata.teams
+				and map.extradata.teams[opponentIndex]
+		end),
+		function(map)
+			return map.extradata.teams[opponentIndex]
+		end
+	)
+
+	local function getSumOf(key)
+		return Array.reduce(Array.map(teamDataPerMap, function (teamData)
+			return teamData[key] or 0
+		end), Operator.add, 0)
+	end
+
+	local postPlant = Array.reduce(teamDataPerMap, function (postPlantTotals, teamData)
+		if teamData.postPlant then
+			postPlantTotals[1] = postPlantTotals[1] + (teamData.postPlant[1] or 0)
+			postPlantTotals[2] = postPlantTotals[2] + (teamData.postPlant[2] or 0)
+		end
+		return postPlantTotals
+	end, {0, 0})
+
+	return {
+		firstKills = getSumOf('firstKills'),
+		thrifties = getSumOf('thrifties'),
+		clutches = getSumOf('clutches'),
+		postPlant = postPlant,
+	}
+end
+
+---@param maps table[]
+---@param player table
+---@param teamIdx integer
+---@return table
+function MatchFunctions.calculateOverallStatsForPlayer(maps, player, teamIdx)
+	local playerId = player.name
+	if not playerId then return {} end
+
+	local overallStats = {
+		acs = {},
+		kast = {},
+		adr = {},
+		kills = 0,
+		deaths = 0,
+		assists = 0,
+		firstKills = 0,
+		firstDeaths = 0,
+		totalRoundsPlayed = 0,
+		totalKastRounds = 0,
+		damageDealt = 0,
+	}
+	local agents = {}
+
+	Array.forEach(maps, function(map)
+		if map.status == MatchGroupInputUtil.MATCH_STATUS.NOT_PLAYED then
+			return
+		end
+
+		local mapOpponent = map.opponents[teamIdx]
+		if not mapOpponent or not mapOpponent.players then
+			return
+		end
+
+		local mapPlayerIndex = Array.indexOf(mapOpponent.players, function(playerData)
+			return playerData.player == playerId
+		end)
+		local mapPlayer = mapOpponent.players[mapPlayerIndex]
+
+		if not mapPlayer then
+			return
+		end
+
+		if mapPlayer.agent then
+			table.insert(agents, mapPlayer.agent)
+		end
+		if mapPlayer.acs then table.insert(overallStats.acs, mapPlayer.acs) end
+		if mapPlayer.kast then table.insert(overallStats.kast, mapPlayer.kast) end
+		if mapPlayer.adr then table.insert(overallStats.adr, mapPlayer.adr) end
+		overallStats.kills = overallStats.kills + (mapPlayer.kills or 0)
+		overallStats.deaths = overallStats.deaths + (mapPlayer.deaths or 0)
+		overallStats.assists = overallStats.assists + (mapPlayer.assists or 0)
+		overallStats.totalRoundsPlayed = overallStats.totalRoundsPlayed + (mapPlayer.totalRounds or 0)
+		overallStats.totalKastRounds = overallStats.totalKastRounds + (mapPlayer.kastRounds or 0)
+		overallStats.damageDealt = overallStats.damageDealt + (mapPlayer.damageDealt or 0)
+
+		local extraDataPlayer = Array.find(map.extradata.teams[teamIdx].players, function(playerData)
+			return playerData.player == playerId
+		end)
+		if not extraDataPlayer then
+			extraDataPlayer = map.extradata.teams[teamIdx].players[mapPlayerIndex]
+		end
+
+		overallStats.firstKills = overallStats.firstKills + (extraDataPlayer.firstKills or 0)
```

Comment:
Why are we storing firstKills in the extradata instead of with the stats on the map level?